awx_collection/plugins/modules/tower_job_wait.py

```diff
@@ -90,78 +92,105 @@
         timeout=dict(type='int'),
         min_interval=dict(type='float', default=1),
         max_interval=dict(type='float', default=30),
     )
 
-    module = TowerModule(
-        argument_spec,
-        supports_check_mode=True
-    )
-
-    json_output = {}
-    fail_json = None
-
-    tower_auth = tower_auth_config(module)
-    with settings.runtime_values(**tower_auth):
-        tower_check_mode(module)
-        job = tower_cli.get_resource('job')
-        params = module.params.copy()
-
-        # tower-cli gets very noisy when monitoring.
-        # We pass in our our outfile to suppress the out during our monitor call.
-        if PY2:
-            outfile = getwriter('utf-8')(StringIO())
-        else:
-            outfile = StringIO()
-        params['outfile'] = outfile
-
-        job_id = params.get('job_id')
-        try:
-            result = job.monitor(job_id, **params)
-        except exc.Timeout:
-            result = job.status(job_id)
-            result['id'] = job_id
-            json_output['msg'] = 'Timeout waiting for job to finish.'
-            json_output['timeout'] = True
-        except exc.NotFound as excinfo:
-            fail_json = dict(msg='Unable to wait, no job_id {0} found: {1}'.format(job_id, excinfo), changed=False)
-        except exc.JobFailure as excinfo:
-            fail_json = dict(msg='Job with id={0} failed, error: {1}'.format(job_id, excinfo))
-            fail_json['success'] = False
-            result = job.get(job_id)
-            for k in ('id', 'status', 'elapsed', 'started', 'finished'):
-                fail_json[k] = result.get(k)
-        except (exc.ConnectionError, exc.BadRequest, exc.AuthError) as excinfo:
-            fail_json = dict(msg='Unable to wait for job: {0}'.format(excinfo), changed=False)
-
-    if fail_json is not None:
-        module.fail_json(**fail_json)
-
-    json_output['success'] = True
-    for k in ('id', 'status', 'elapsed', 'started', 'finished'):
-        json_output[k] = result.get(k)
-
-    module.exit_json(**json_output)
+    # Create a module for ourselves
+    module = TowerModule(argument_spec=argument_spec, supports_check_mode=True)
+
+    # Extract our parameters
+    job_id = module.params.get('job_id')
+    timeout = module.params.get('timeout')
+    min_interval = module.params.get('min_interval')
+    max_interval = module.params.get('max_interval')
+
+    # Attempt to look up job based on the provided id
+    job = module.get_one('jobs', **{
+        'data': {
+            'id': job_id,
+        }
+    })
+
+    if job is None:
+        module.fail_json(msg='Unable to wait, on job {0} that ID does not exist in Tower.'.format(job_id))
+
+    job_url = job['url']
+
+    # This comes from tower_cli/models/base.py from the old tower-cli
+    dots = itertools.cycle([0, 1, 2, 3])
+    interval = min_interval
+    start = time.time()
+
+    # Poll the Ansible Tower instance for status, and print the status to the outfile (usually standard out).
```

Comment:
the check_job method doesn't seem to poll

Reply:
Fixed the comments.